Refactor to have references in separate sub-directory; closes #49 (#77)
astrodbtoolkit/AstrodbKit2: docs/index.rst

--------------------

**Astrodbkit2** contains methods to output the full contents of the database as a list of JSON files.

> It can likewise read in a directory of these files to populate the database. 
> By default, reference tables (eg, Publications, Telescopes, etc) are stored in a `reference` sub-directory 
> and the source tables are in a `source` sub-directory. 

Reviewer: ```suggestion
It can likewise read in a directory of these files to populate the database. 
By default, reference tables (eg, Publications, Telescopes, etc) and source tables are respectively stored in `reference/` and `source/` sub-directories of `data/`.
```

===================

If users perform changes to a database, they will want to output this to disk to be version controlled.

> **Astrodbkit2** provides methods to save an individual source or reference table as well as the entire data. 
> By default, reference tables are stored in a sub-directory called "reference"; this can be overwritten by 
> supplying a `reference_directory` variable into `save_database` or `save_reference_table`. 
> Similarly, source/object tables are stored in a sub-directory called "source" which can be overwritten by supplying 
> `source_directory`.

Reviewer: ```suggestion
**Astrodbkit2** provides methods to save an individual source or reference table as well as all of the data stored in the database. 
By default, reference tables are stored in a sub-directory of `data/` called "reference"; this can be overwritten by 
supplying a `reference_directory` variable into `save_database` or `save_reference_table`. 
Similarly, source/object tables are stored in a sub-directory of `data/` called "source" which can be overwritten by supplying  a `source_directory` variable.
```

    db.save_reference_table('Publications', 'data')

>     # Save entire database to directory 'data'

Reviewer: ```suggestion
    # Save entire database to directory 'data/' with 'reference/' and 'source/' subdirectories.
```
